web/src/app/people/person-files-form/person-files-form.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { TreeService } from 'src/app/tree/shared/tree.service';
+import { FileResponse } from '../shared/file-response.model';
+import { PeopleService } from '../shared/people.service';
+import { RemovePersonsFileRequest } from '../shared/remove-persons-file.model';
+
+@Component({
+  selector: 'app-person-files-form',
+  templateUrl: './person-files-form.component.html',
+  styleUrls: ['./person-files-form.component.scss']
+})
+export class PersonFilesFormComponent implements OnInit {
+  treeId: string;
+  personId: string;
+  mainPhotoUrl: string;
+  files: FileResponse[];
+
+  constructor(
+    private service: PeopleService,
+    private treeService: TreeService,
+    private route: ActivatedRoute
+  ) { }
+
+  ngOnInit(): void {
+    this.treeId = this.route.parent.snapshot.paramMap.get('id');
+    this.personId = this.route.snapshot.paramMap.get('id');
+
+    this.treeService.getTree(this.treeId).subscribe(tree => {
+      const person = tree.people.find(per => per.id == this.personId);
+      if(person.mainPhoto != null) {
+        this.mainPhotoUrl = person.mainPhoto.uri;
+      }
+      this.files = person.files;
+    });
+  }
+
+  uploadPhoto(files: File[]) {
+    const formData = this.prepareFormData(files[0]);
```

Comment:
Wouldn't it throw on button press when No file selected?
(Just asking, maybe the button is grayed out)

Reply:
File is upload automatically when user open it. There is no confirm button